### ISSUE TITLE
Compatibility with web extensions (Firefox 57)

### ISSUE THREAD

**Issue by damienfournier** [gmail-manager-community/gmail-manager-community#48]
Hi,

Firefox will change its plugin system in Firefox 52, killing the old way to support only web extensions.

Is gmail manager compatible ?
Does it need to be rewritten ?

**Comment by ath0mas** [comment]
currently not a webExtension 😞

**Comment by ath0mas** [comment]
> It will be the only type of add-on supported in Firefox by the end of 2017 with the release of Firefox 57.

* https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Porting_a_legacy_Firefox_add-on
* https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Comparison_with_XUL_XPCOM_extensions

**Comment by damienfournier** [comment]
Do you think you can do it ?
Wouldn't be the time to try using apis instead of regexing the gmail page ?

**Comment by damienfournier** [comment]
Hello,

any news on this ?
It's better if we can sort this issue now, or gmanager will stop suddenly as soon as firefox 57 is out..

**Comment by Catalyst2012** [comment]
Thanks to all the developers!

I am a grateful user with several Gmail addresses who will miss this great extension after Firefox 56. Its potential loss is one of the primary reasons I am experimenting with other solutions such as using Chrome and Opera. Gmail Manager Community is better than any extensions I've seen over there and still works well in Pale Moon (which I am using now) but who knows for how long.

For security reasons, its wise to keep one's browser(s) updated. That won't be possible for me after  Firefox 56 as I am not willing to lose all the great add-ons I've learned to trust over the years.

So I am requesting the possibility of porting this extension to the Chromium-based browser family. Then we (hopefully) won't have to deal with Mozilla's ever-changing landscape. After all, who knows when Mozilla will change it again.

**Comment by damienfournier** [comment]
@Catalyst2012 
If it's rewritten as a web extension, it should be compatible with chrome (from what I've understood).
The problem is: nobody here have the time/skills to rewrite the extension, since the original authors are MIA for years...
But if someone can do that, that would be great !

**Comment by damienfournier** [comment]
@m-soltys you are our last hope :/ 

2-3 weeks before the end ?

**Comment by m-soltys** [comment]
Hi guys, 

I'll try to find out some time over the weekend and take a look on this compatibility thing.

**Comment by Catalyst2012** [comment]
YES, I would greatly appreciate this also, and thanks again to all of the great developers!!!

For those that may be desperate, may I suggest two options: 1) Don't upgrade to Firefox 57, and 2) try PaleMoon; my understanding is they will continue to allow Classic (Legacy) extensions. At present,  [Palemoon](https://www.palemoon.org/) is significantly faster at loading pages than Firefox, and hangs less.

**Comment by damienfournier** [comment]
Thanks @m-soltys , can we help you on this ?

For information, the release date of firefox 57 is 2017-11-14

**Comment by m-soltys** [comment]
I looked at this compatibility thing and I don't have good news. It's not a small and easy change, it isn't even medium. Extension need to be practically rewritten as basics how the extensions work will change in the next version of the Firefox.

I will slowly work on this. But there are couple 'buts' - I'm not a JS developer, I don't know Firefox SDK and I don't have much time lately so this will take me some time. 

I'm staying with Firefox 56 anyways because I'm using bunch of extensions that aren't ported to the web extensions and I noticed that only one sidebar could be open at a time in the new version of Firefox which I don't like at all.
